### ISSUE TITLE
JS: small issue with Std.is and Firefox

### ISSUE THREAD

**Issue by mockey** [HaxeFoundation/haxe#2297]
Std.is(elem, Element) returns false for an Element in Firefox, because
```typeof Element```
returns "object" in Firefox while it returns "function" in WebKit browsers. Modifying the check in ```js.Boot.__instanceof``` to:
```
if( (untyped __js__("typeof"))(cl) == "function" || (untyped __js__("typeof"))(cl) == "object" ) {
  ...
```
seems to fix this.

**Comment by aduros** [comment]
Too late for 3.2, possible for 3.3.

**Comment by mockey** [comment]
A fix from me is already merged since last year: #2857. This can be closed.
